src/DataTypes/Serializations/SerializationTuple.cpp

```diff
@@ -44,9 +44,11 @@ void SerializationTuple::deserializeBinary(Field & field, ReadBuffer & istr) con
 {
     const size_t size = elems.size();
 
-    Tuple tuple(size);
+    field = Tuple();
+    Tuple & tuple = get<Tuple &>(field);
+    tuple.reserve(size);
     for (const auto i : collections::range(0, size))
-        elems[i]->deserializeBinary(tuple[i], istr);
+        elems[i]->deserializeBinary(tuple.emplace_back(), istr);
 
     field = tuple;
```

Comment:
Consider to use `std::move` or remove this line.

Reply:
Good catch.